aesm-client/src/imp/sgx.rs

```diff
@@ -0,0 +1,129 @@
+use std::net::TcpStream;
+use std::io::{Read,Write};
+
+use byteorder::{LittleEndian, NativeEndian, ReadBytesExt, WriteBytesExt};
+use protobuf::Message;
+
+pub use error::{AesmError, Error, Result};
+use {
+    AesmRequest, FromResponse, QuoteInfo, QuoteResult, QuoteType,
+    Request_GetQuoteRequest, Request_InitQuoteRequest,
+};
+
+/*
+// FIXME: uncomment this after resolving https://github.com/fortanix/rust-sgx/issues/31
+use std::time::Duration;
+/// This timeout is an argument in AESM request protobufs.
+///
+/// This value should be used for requests that can be completed locally, i.e.
+/// without network interaction.
+const LOCAL_AESM_TIMEOUT_US: u32 = 1_000_000;
+/// This timeout is an argument in AESM request protobufs.
+///
+/// This value should be used for requests that might need interaction with
+/// remote servers, such as provisioning EPID.
+const REMOTE_AESM_TIMEOUT_US: u32 = 30_000_000;
+*/
+
+#[derive(Debug)]
+pub struct AesmClient {
+    tcp_stream: TcpStream,
+}
+
+impl AesmClient {
```

Comment:
A lot of this code is shared with unix, it would be good to use a common implementation somehow. i.e. create another `mod aesm_protobuf` or so that you depend on in both unix and sgx.